sample.js

```diff
@@ -1,9 +1,9 @@
-var yahooWeatherShim = require('./src/yahooWeather.js');
+var yahooWeather = require('./src/yahooWeather.js');
 var place = "New Delhi";
 
-var weatherData = new yahooWeatherShim(place);
+var weather = new yahooWeather();
 
-weatherData.getFullWeather().then(function(res){
-	weatherData.init(res);
-	console.log(weatherData.getForecast());
+weather.init(place).then(function(weatherData){
+	weather.setWeatherData(weatherData);
```

Comment:
Usually when you are using a Proxy class to fetch data for your user code, it will store the data inside automatically. This benefits the program from the aspect of caching and convenience. Therefore, if the task of `init`is to fetching data, it can call the `setWeatherData` before it pass the data out to follow the same pattern. This is also preventing that user forgot to update the fetched one like in this sample code does.